src/Tinker.jl

```diff
@@ -121,360 +131,77 @@ function drawrecthandle(ctx, rh::RectHandle, d, color1, width)
     end
 end
 
-# Given a RectHandle and a position, returns the corresponding Handle
-function get_handle(rh::RectHandle, pos::String)
-    for i in 1:8
-        pos == rh.h[i].pos && return rh.h[i]
+function drawline(ctx, l, color, width)
+    isempty(l) && return
+    p = first(l)
+    move_to(ctx, p.x, p.y) 
+    set_source(ctx, color)
+    set_line_width(ctx, width)
+    for i = 2:length(l)
+        p = l[i] 
+        line_to(ctx, p.x, p.y)
     end
-    println("Not a valid position")
-    return Handle()
+    stroke(ctx)
 end
 
-# Gets anchor point for modification by handle
-function get_p1(h::Handle, rh::RectHandle)
-    opposite_dict = Dict('b'=>"t", 't'=>"b", 'l'=>"r", 'r'=>"l")
-    if h.pos[end] == 'c' # 'h' is a corner
-        opp_pos = ""
-        # build pos string for opposite corner
-        opp_pos *= opposite_dict[h.pos[1]]
-        opp_pos *= opposite_dict[h.pos[2]]
-        opp_pos *= "c"
-        # set p1
-        opposite = get_handle(rh, opp_pos) # Handle object
-        p1 = XY{UserUnit}(opposite.x,opposite.y) # XY object
-    elseif h.pos[end] == 's' # 'h' is a side
-        if h.pos == "ts"
-            # modify top
-            p1 = XY{UserUnit}(rh.h[5].x,rh.h[5].y)
-        elseif h.pos == "rs"
-            # modify right
-            p1 = XY{UserUnit}(rh.r.x, rh.r.y)
-        elseif h.pos == "bs"
-            # modify bottom
-            p1 = XY{UserUnit}(rh.r.x, rh.r.y)
-        elseif h.pos == "ls"
-            # modify left
-            p1 = XY{UserUnit}(rh.h[5].x,rh.h[5].y)
-        end
-    else
-        println("Invalid position in get_p1")
-        p1 = XY{UserUnit}(0,0)
+function ispolygon(p::AbstractArray) # type better
+    length(p) < 3 && return false
+    for i in 1:(length(p)-1)
+        p[i]!=p[i+1] && return true
     end
-    return p1
+    return false
```

Comment:
Is this implicitly closed, or do you want to check/insist that `p[end] == p[1]`?